From b5cc964fdec048ae84f0293a0277eae26f850b3c Mon Sep 17 00:00:00 2001
From: Tomasz Duszynski <tduszynski@cavium.com>
Date: Thu, 14 Mar 2019 15:55:26 +0100
Subject: [PATCH 047/386] octeontx2-af: annotate flow_msg fields with
 endian-aware types

Change variables types to endian-aware ones so that sparse could help
with catching endianness errors.

Change-Id: I17faed9758449bda7eb52a2da92d6a502ac9e323
Signed-off-by: Tomasz Duszynski <tduszynski@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5713
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/af/mbox.h  | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
index 850b1a62a176..de0a78c11d56 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
@@ -1012,27 +1012,26 @@ enum header_fields {
 	NPC_HEADER_FIELDS_MAX,
 };
 
-/* all multi byte fields should be in network byte order */
 struct flow_msg {
 	unsigned char dmac[6];
 	unsigned char smac[6];
-	u16 etype;
-	u16 vlan_etype;
-	u16 vlan_tci;
+	__be16 etype;
+	__be16 vlan_etype;
+	__be16 vlan_tci;
 	union {
-		u32 ip4src;
-		u32 ip6src[4];
+		__be32 ip4src;
+		__be32 ip6src[4];
 	};
 	union {
-		u32 ip4dst;
-		u32 ip6dst[4];
+		__be32 ip4dst;
+		__be32 ip6dst[4];
 	};
 	u8 tos;
 	u8 ip_ver;
 	u8 ip_proto;
 	u8 tc;
-	u16 sport;
-	u16 dport;
+	__be16 sport;
+	__be16 dport;
 };
 
 struct npc_install_flow_req {
-- 
2.17.1

